ami/flowchart/library/Operators.py

```diff
@@ -633,7 +633,7 @@ def __init__(self, name):
 
         def display(self, topics, terms, addr, win, **kwargs):
             if self.widget is None:
-                self.widget = FilterWidget(terms, self.output_vars(), win)
+                self.widget = FilterWidget(terms or self.input_vars(), self.output_vars(), win)
```

Comment:
This is the fix for the filter. The state is getting out of sync between the flowchart and the subprocess. The subprocess calls display with all the arguments as none, but the subprocess has them filled in, which causes the inconsistency in state. 

Maybe a better fix would be to call display with the terms argument in build_views.